src/main/java/mate/academy/spring/validator/EmailValidator.java

```diff
@@ -0,0 +1,14 @@
+package mate.academy.spring.validator;
+
+import java.util.regex.Pattern;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class EmailValidator implements ConstraintValidator<Email, String> {
+    private static final String EMAIL_PATTERN = "^[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,6}$";
+
+    @Override
+    public boolean isValid(String email, ConstraintValidatorContext constraintValidatorContext) {
+        return email != null && Pattern.compile(EMAIL_PATTERN).matcher(email).matches();
```

Comment:
Let's compile the string pattern only once. There is no need to do it each time this method is called